neural_compressor/adaptor/torch_utils/weight_only.py

```diff
@@ -122,9 +183,307 @@ def rtn_quantize(model, num_bits, group_size=-1, scheme="asym", w_layers_config=
             weight = weight.reshape(weight.shape[0], -1)
         else:
             weight = m.weight
-        q_weight = quant_weight(weight, num_bits, group_size, scheme)
+        q_weight = quant_weight(weight, num_bits, group_size, scheme, quantile)
         if m.__class__.__name__ == "Conv2d":
             q_weight = q_weight.reshape(orig_shape[1], orig_shape[0], orig_shape[2], orig_shape[3])
             q_weight = q_weight.permute(1, 0, 2, 3)
         m.weight.data.copy_(q_weight)
     return model
+
+
+def get_module_input_output(model, module_hook_config={}, dataloader=None, iters=-1, 
+                            calib_func=None):
+    """A help function to get input and output tensor of modules in module_name_list.
+
+    Args:
+        model: torch model.
+        module_hook_config (dict, optional): required module name for input/output. Defaults to {}.
+            For example:
+                module_hook_config = {
+                    'fc1': ['output'],
+                    'fc2': ['input', 'output']
+                }
+        dataloader: dataloader for model input.
+        iters: iterations for inference.
+        calib_func: a custom inference function to replace dataloader and iters.
+
+    Returns:
+        input_values, output_values: recorded input_values, output_values.
+    """
+    input_values, output_values = {}, {}
+    def _save_input_output_hook(name):
+        """
+        A forward hook to save input and output values of a module
+            param name: the module name
+            return: A hook function
+        """
+        def save_input_hook(module, inputs):
+            input = inputs[0]
+            if name in input_values:
+                input_values[name] = torch.cat((input_values[name], input), 0)
+            else:
+                input_values[name] = input
+        def save_output_hook(module, inputs, outputs):
+            if isinstance(outputs, tuple):
+                outputs = outputs[0]
+            if name in output_values:
+                output_values[name] = torch.cat((output_values[name], outputs), 0)
```

Comment:
How many samples will it save?  The memory requirement of the intermediate output will be huge

Reply:
I will only collect 10 modules each time in awq_quantize caibration func.